src/sync/sync-utils.ts

```diff
@@ -3,21 +3,20 @@ import { sqsQueues } from "../sqs/queues";
 import { Subscription, SyncStatus } from "models/subscription";
 import Logger from "bunyan";
 import { numberFlag, NumberFlags } from "config/feature-flags";
-import { TaskType } from "~/src/sync/sync.types";
+import { TaskType, SyncType } from "~/src/sync/sync.types";
 import { GitHubAppConfig } from "~/src/sqs/sqs.types";
 import { envVars } from "config/env";
 import { GitHubServerApp } from "models/github-server-app";
 import { GITHUB_CLOUD_API_BASEURL, GITHUB_CLOUD_BASEURL } from "~/src/github/client/github-client-constants";
 
-type SyncType = "full" | "partial";
-
 export const findOrStartSync = async (
 	subscription: Subscription,
 	logger: Logger,
-	syncType?: SyncType,
-	commitsFromDate?: Date,
+	syncType: SyncType,
+	commitsFromDate: Date | undefined,
```

Comment:
I think previous typing were better. Any advantage of using `| undefined`  over optional?

Reply:
For the `syncType`, mandatory is always better than optional.
For `commitsFromDate`, I'd love to make it mandatory as well. But the logic change is a bit more, so leave it til later.

----------
Meanwhile, it is less bug prone to have explicit param (but can be undfined) compared to optional params. Think about `gitHubAppId`, 
calling a function `getServerApp(someJiraHost, gitHubAppId<--which is undefined)` is better to calling a function that can be `getServerApp(someJiraHost)`. Because for the second one, it could be a bug that someone forgot to passin the `gitHubServerAppId` which may or may not be undefined.

Reply:
But it depends. I can still accept that for truly `optional` param, we take it as `optional`. Just mind that when a function has, say, 4 option params, like `doSomething(a, b?, c?, d?)`, it is quite in flexible, as in order to pass in `c`, you need to pass in `b` first. 

So I prefer using `object`, like
```
findOrStartSync({
  subscription,
  syncType: 'partial',
  commitsFromDate: undefined
});
```

However that's not how the rest of our code looks like. So keep it consistent for this one.